### PR TITLE
Add README for bor module

### PR DESCRIPTION
**Pull request by Raneet10**
# Description

This PR adds a `README` for `bor` module.

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Changes only for a subset of nodes

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [x] I have added sufficient documentation in code
- [x] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

## Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [ ] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### CODE REVIEW

File: bor/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# PRELIMINARY TERMINOLOGY
+
+* A `side-transaction` is a normal heimdall transaction but the data with which the message is composed needs to be voted on by the validators since the data is obscure to the consensus protocol itself and it has no way of validating the data's correctness.
+* A `sprint` comprises of 16 blocks.
```
Comment (manav2401) [inline bor/README.md:4]: Do you think it's good to mention 16 bor blocks here?

File: bor/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+* `propose-span` - Print the `propose-span` command :
+
+via CLI
+```
+heimdallcli query bor propose-span --proposer <VALIDATOR ADDRESS> --start-block <START_BLOCK> --span-id <SPAN_ID> --bor-chain-id <BOR_CHAIN_ID>
+```
+
+via Rest
+```
+curl "localhost:1317/bor/prepare-next-span?span_id=23&start_block=1000&chain_id="137""
```
Comment (manav2401) [inline bor/README.md:165]: Start block and chain id are hardcoded here. Is it intentional?

File: bor/README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+* `span` - Query the span corresponding to the given span id:
+
+via CLI
+```
+heimdallcli query bor span --span-id=<span ID here>
+```
+
+via Rest
+```
+curl localhost:1317/bor/span/1000
```
Comment (manav2401) [inline bor/README.md:110]: Span id is hardcoded here. Is it intentional?

**Review by manav2401 — APPROVED** [review]
Noice, LGTM. Just a few suggestions.

File: bor/README.md (hunk 2 above)
Comment (Raneet10) [reply bor/README.md:165]: Maybe. I was probably just caught in two minds whether to add an example or a generalized query 😄 
I'll rectify it, thanks!

File: bor/README.md (hunk 1 above)
Comment (Raneet10) [reply bor/README.md]: Addressed [here](https://github.com/maticnetwork/heimdall/pull/1071/commits/4fc537d495cb854277cacff52c1b4a6282ff0bdf)

File: bor/README.md (hunk 3 above)
Comment (Raneet10) [reply bor/README.md]: Rectified [4fc537d](https://github.com/maticnetwork/heimdall/pull/1071/commits/4fc537d495cb854277cacff52c1b4a6282ff0bdf).

File: bor/README.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+```
+err := k.FreezeSet(ctx, msg.ID, msg.StartBlock, msg.EndBlock, msg.ChainID, msg.Seed)
+if err != nil {
+	k.Logger(ctx).Error("Unable to freeze validator set for span", "Error", err)
+	return common.ErrUnableToFreezeValSet(k.Codespace()).Result()
+}
+```
+
+`FreezeSet` internally invokes `SelectNextProducers`, which pseudo-randomly picks producers from the validaor set, leaning more towards validators with higher stake:
+
```
Comment (marcello33) [inline bor/README.md:59]: shall we say higher voting power based on stake instead of "higher stake"?

File: bor/README.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
+# PRELIMINARY TERMINOLOGY
```
Comment (marcello33) [inline bor/README.md:1]: I'd prefer all lower case except first letter, but let's find a convention with @0xKrishna for all other modules

File: bor/README.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+One can run the following query commands from the bor module :
+
+* `span` - Query the span corresponding to the given span id:
+
+via CLI
+```
+heimdallcli query bor span --span-id=<SPAN_ID>
+```
+
+via Rest
```
Comment (marcello33) [inline bor/README.md:108]: let's use REST (all capital letters here :D )

File: bor/README.md (hunk 5 above)
Comment (Raneet10) [reply bor/README.md:1]: I'm okay with either. But yeah, will wait for @0xKrishna to make the final call.

File: bor/README.md (hunk 6 above)
Comment (Raneet10) [reply bor/README.md]: Done [here](https://github.com/maticnetwork/heimdall/pull/1071/commits/a0e215b352f023586dc177fe200a57f6e388a40f).

File: bor/README.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
+# PRELIMINARY TERMINOLOGY
+
+* A `side-transaction` is a normal heimdall transaction but the data with which the message is composed needs to be voted on by the validators since the data is obscure to the consensus protocol itself and it has no way of validating the data's correctness.
+* A `sprint` comprises of 16 bor blocks.
+* A `span` comprises of 100 sprints in bor.
```
Comment (0xKrishna) [inline bor/README.md:5]: I think it's 400 now with 16 block sprint, Also sprint is decided by bor and not heimdall.

File: bor/README.md (hunk 5 above)
Comment (0xKrishna) [reply bor/README.md:1]: I agree with @marcello33

File: bor/README.md (hunk 7 above)
Comment (Raneet10) [reply bor/README.md]: Added in [this](https://github.com/maticnetwork/heimdall/pull/1071/commits/096779cf5bb71ba835d04dd5fa866515ed045a6a) commit.

File: bor/README.md (hunk 5 above)
Comment (Raneet10) [reply bor/README.md]: Done [here](https://github.com/maticnetwork/heimdall/pull/1071/commits/096779cf5bb71ba835d04dd5fa866515ed045a6a).

**Review by 0xKrishna — APPROVED** [review]

**Pull request merged**